### ISSUE TITLE
Use image "title" attribute for captions rather than "alt" attribute

### ISSUE THREAD

**Issue by rootwork** [chipzoller/hugo-clarity#219]
(Omitting issue template since this is more of a feature request than a bug report.)

Unless I'm mistaken, [this change](https://github.com/chipzoller/hugo-clarity/pull/206/commits/d1a105457e5451e01dfe2ecf41478b80949612bc) to #206 reverted to using the `alt` attribute in images to populate captions, rather than `title`, which is what I'd originally proposed in that PR (mentioned in item 3). Don't get me wrong, I'm really excited you accepted what I know was an enormous PR! And of course it's your decision whether to continue using `alt` rather than `title` for captions, but I want to make the case for why I think it should be the latter.

Just to be sure we're on the same page, an image in Markdown defines its attributes this way:
```md
![alt text](/path/to/img.jpg "title text")
```

On using captions, [WebAIM](https://webaim.org/techniques/alttext/#figure) says it succinctly:

> figure and figcaption allow a semantic association between a figure and the figure's caption. The figcaption may provide a summary of or additional information about the figure and/or relate the figure back to the containing document. However, any image within a figure must still have an alt attribute value that presents the alternative text of the image - **this should not be conveyed via the figcaption**.

(Emphasis mine.)

Other useful summaries making this same argument:
- "[Web Accessibility: Images](https://www.brandeis.edu/web-accessibility/understanding/images-alt-text.html)"
- "[Image ALT Text](https://accessibility.psu.edu/images/alttext/)"
- "[How to Create Alternative Text for Images for Accessibility and SEO](https://www.techsmith.com/blog/how-to-create-alternative-text-for-images-for-accessibility-and-seo/)"

So alt text isn't really intended to be the same as a caption. And conversely, even if you don't want to display a caption, you _should_ provide alt text unless it's a purely [decorative image](https://www.w3.org/WAI/tutorials/images/decorative/).

### Summary

Deriving captions from `title` rather than `alt`:

- is better for accessibility
- gives authors more options in separately defining each

### Drawbacks

The one obvious downside is that sites that had been using `alt`s for captions until now would suddenly have empty captions if you switched to basing it on `title`. I'd be up for coming up with a regex script to duplicate `alt` into `title` where `title` doesn't already exist, if that would help assuage that concern.

I suppose it's also a bit more work to type
```md
![alt text, possibly empty](/path/to/img.jpg "intended caption text")
```
rather than
```md
![intended caption text](/path/to/img.jpg)
```
but I think that's outweighed by the positives.

**Comment by chipzoller** [comment]
I think you make a good case and I would like us to align on best practices in the industry. Although I would have to personally go to all of my blog posts (since I use this theme) and make the change, I'm glad to do so. But other than a practical recommendation, what is your suggestion here? Is this in need of code or just documentation?

**Comment by rootwork** [comment]
The code piece is that the JS is grabbing the content of the `alt` attribute to populate the captions, so that would have to change to grabbing `title` instead. I can pull that part out from the original version of the PR and create a new PR if you agree.

Do you want to include some sort of script to auto-duplicate `alt`s to empty `title`s, or is that too much?

**Comment by chipzoller** [comment]
cc @onweru

**Comment by onweru** [comment]
@rootwork, you make a good point. I noticed the change when I reverted but wasn't aware exactly why you did it. Umh, I would suggest that we proceed as follows. 

1. If the title attribute is provided, use it as the caption
2. Else, use the alt as the caption

Option 1 is of course the ideal, because it follows the industry spec. However, not everyone is aware. I reckon most people don't even add a title attribute in their markdown.

This approach would be backwards compatible.

For the actual implementation, the alt would be the default and the `title` would be used only where it's defined

```javascript
let caption = image.title ? image.title : image.alt;
```
